### PR TITLE
W22 5pm 3 ig implement current day and end date in commonsOverview

### PR DESCRIPTION
**Pull request by isaiahgarcia**
# Overview

This PR creates a `startDate` and `duration` variable to `Commons.java`. I add these as parameters to the Create Commons Swagger API in `CommonsController.java` as well as implement UI Form Groups in `CreateCommonsForm.js`. I also use the moment library to retrieve the current day and calculate the end date based off the start date and duration of the game. The reason why this PR is created is because we want to display the current date and the last day of the game.

# Issues Addressed
#38 

# Details
Coverage:
Screen Shot 2022-03-11 at 4 51 30 AM

Mutation Tests for CommonsOverviewTest.js:
Screen Shot 2022-03-11 at 4 49 50 AM

### CODE REVIEW

File: frontend/src/main/components/Commons/CommonsOverview.js [hunk 1]
```diff
@@ -1,14 +1,29 @@
 import React from "react";
 import { Card } from "react-bootstrap";
+import moment from "moment";
```
Comment (pconrad) [inline frontend/src/main/components/Commons/CommonsOverview.js:3]: I've asked that the dependency on `moment.js` be removed.

See: https://github.com/ucsb-cs156-w22/team04-w22-5pm-HappyCows/issues/59

File: frontend/src/main/components/Commons/CommonsOverview.js [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+
+function getStartDate(timestamp) {
+    return moment(timestamp).format('YYYY-MM-DD');
```
Comment (pconrad) [inline frontend/src/main/components/Commons/CommonsOverview.js:6]: Please do this with native JS functions, or with a more modern JS library such as date-fns or luxon.  See: 

https://momentjs.com/docs/#/-project-status/

And: 

* [You Probably Don't Need Moment.js Anymore](https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore)
 * [You don't (may not) need Moment.js](https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md)
 * [Why you shouldn't use Moment.js...](https://inventi.studio/en/blog/why-you-shouldnt-use-moment-js)
 * [4 alternatives to moment.js for internationalizing dates](https://blog.logrocket.com/4-alternatives-to-moment-js-for-internationalizing-dates/)

File: frontend/src/tests/components/Commons/CommonsOverview.test.js [hunk 3]
```diff
@@ -1,12 +1,35 @@
 import { render } from "@testing-library/react";
-import CommonsOverview from "main/components/Commons/CommonsOverview"; 
+import CommonsOverview from "main/components/Commons/CommonsOverview";
+import { getStartDate, getEndDate } from "main/components/Commons/CommonsOverview";  
 import commonsFixtures from "fixtures/commonsFixtures"; 
+import moment from "moment";
```
Comment (pconrad) [inline frontend/src/tests/components/Commons/CommonsOverview.test.js:5]: See above about moment.js

**Review by pconrad — CHANGES_REQUESTED** [review]
Please try to remove the dependency on moment.js

File: frontend/src/main/components/Commons/CommonsOverview.js (hunk 2 above)
Comment (tekaireb) [reply frontend/src/main/components/Commons/CommonsOverview.js:6]: @isaiahgarcia I have already created a pure / vanilla JavaScript function to convert timestamps into the desired date format. Feel free to use it. The relevant code is currently awaiting review in [pull request #62](https://github.com/ucsb-cs156-w22/team04-w22-5pm-HappyCows/pull/62).

**Review by tekaireb — CHANGES_REQUESTED** [review]
Looks good except for the usage of `moment.js`. Professor Conrad very recently mentioned that it has been deprecated. However, I have already created a function which you may use to format dates, or you can implement your own if you'd prefer.

**Pull request closed without merging**